### PR TITLE
Code cleanup of Hearing module

### PR DESCRIPTION
**Pull request by jokoho48**
Code cleanup of Hearing module.

### CODE REVIEW

File: addons/hearing/functions/fnc_firedNear.sqf [hunk 1]
```diff
@@ -37,11 +37,11 @@ _vehAttenuation = if ((ACE_player == (vehicle ACE_player)) || {isTurnedOut ACE_p
 
 _distance = 1 max _distance;
 
-_silencer = switch (_weapon) do {
-    case (primaryWeapon _firer) : {(primaryWeaponItems _firer) select 0};
-    case (secondaryWeapon _firer) : {(secondaryWeaponItems _firer) select 0};
-    case (handgunWeapon _firer) : {(handgunItems _firer) select 0};
-    default {""};
+_silencer = call {
+    if (primaryWeapon _firer) exitWith {(primaryWeaponItems _firer) select 0};
```
Comment (PabstMirror) [inline addons/hearing/functions/fnc_firedNear.sqf]: This has a string inside of a if statement and no comparison to _weapon

File: addons/hearing/functions/fnc_firedNear.sqf [hunk 2]
```diff
@@ -68,11 +68,11 @@ if (count _weaponMagazines == 0) then {
 
 _magazine = "";
 {
-    EXPLODE_2_PVT(_x,_magazineType,_ammoType);
+    params ["_magazineType", "_ammoType"];
```
Comment (PabstMirror) [inline addons/hearing/functions/fnc_firedNear.sqf]: needs `_x` in front

Comment (jokoho48) [inline addons/hearing/functions/fnc_firedNear.sqf]: oh right sry miss that

File: addons/hearing/functions/fnc_firedNear.sqf (hunk 1 above)
Comment (jokoho48) [inline addons/hearing/functions/fnc_firedNear.sqf]: oh yeah i miss that

**Comment by PabstMirror** [comment]
I tested the call/switch statements, I'm not seeing much benifit in bis's benchmark:

```
_firer = player;
_weapon = (handgunWeapon player);
_silencer = switch (_weapon) do {
    case (primaryWeapon _firer) : {(primaryWeaponItems _firer) select 0};
    case (secondaryWeapon _firer) : {(secondaryWeaponItems _firer) select 0};
    case (handgunWeapon _firer) : {(handgunItems _firer) select 0};
    default {""};
};
0.0117188 ms
0.0116699 ms

_firer = player;
_weapon = (handgunWeapon player);
_silencer = call {
    if ((primaryWeapon _firer) == _weapon) exitWith {(primaryWeaponItems _firer) select 0};
    if ((secondaryWeapon _firer) == _weapon) exitWith {(secondaryWeaponItems _firer) select 0};
    if ((handgunWeapon _firer) == _weapon) exitWith {(handgunItems _firer) select 0};
    ""
};
0.012207 ms
0.0123047 ms
```

**Comment by jokoho48** [comment]
yeah right see what you mean hmm ok i try it self

**Comment by Glowbal** [comment]
In fact looking at those tests, it seems to be faster to stick with the `switch`. It also is easier to read/understand without having to take a closer look imo.

**Comment by jokoho48** [comment]
but if you use a swtich True call if exitWith is faster

**Comment by PabstMirror** [comment]
looks good, merging before looking at #1988 to avoid conflicts.

**Pull request merged**